docs/new_packages.md

```diff
@@ -148,3 +148,51 @@ Shell commands to run after building the library. These are run inside of
 A list of required packages.
 
 (Unlike conda, this only supports package names, not versions).
+
+## Manual creation of a Pyodide package
+The previous sections describes how to add a python package to the pyodide
+build.
+
+There are cases where you want to ship additional python libraries without
+adding it to pyodide itself and recompiling everything. For simplicity, let us
+assume the library is pure python.
+
+It is helpful to have some understanding of the structure of a Pyodide package.
+Pyodide is obtained by compiling CPython into web assembly. As such, it loads
+packages the same way as CPython --- it looks for relevant files `.py` files in
+`/lib/python3.x/`. When creating and loading a package, our job is to put our
+`.py` files in the right location in emscripten's virtual filesystem.
+
+Suppose you have a python library that consists of a single directory
+`/PATH/TO/LIB/` whose contents would go into
+`/lib/python3.8/site-packages/PACKAGE_NAME/` under a normal python
+installation.
+
+The simplest version of the corresponding Pyodide package contains two files
+--- `PACKAGE_NAME.data` and `PACKAGE_NAME.js`. The first file
+`PACKAGE_NAME.data` is a concatenation of all contents of `/PATH/TO/LIB`. When
+loading the package via `pyodide.loadPackage`, Pyodide will load and run
+`PACKAGE_NAME.js`. The script then fetches `PACKAGE_NAME.data` and extracts the
+contents to emscripten's virtual filesystem. Afterwards, since the files are
+now in `/lib/python3.8/`, running `import PACKAGE_NAME` in python will
+successfully import the module as usual.
+
+To produce these files, download the `file_packager.py` script from
+[https://github.com/iodide-project/pyodide/blob/master/tools/file_packager.py](https://github.com/iodide-project/pyodide/blob/master/tools/file_packager.py). You then run the command
+```sh
+$ ./file_packager.py PACKAGE_NAME.data --js-output=PACKAGE_NAME.js --abi=1 --export-name=pyodide._module --use-preload-plugins --preload /PATH/TO/LIB/@/lib/python3.8/site-packages/PACKAGE_NAME/ --exclude "*__pycache__*"
+```
+The `--preload` argument instructs the package to look for the file/directory
+before the separator `@` (namely `/PATH/TO/LIB/`) and place it at the path
+after the `@` in the virtual filesystem (namely
+`/lib/python3.8/site-packages/PACKAGE_NAME/`). Remember to use the correct python version in the target path. At the time of writing, the latest release of Pyodide uses python 3.7 while git master uses python 3.8.
+
+The `--exclude` argument
+specifies files to omit from the package. This argument can be repeated, e.g.
+you can append `--exclude README.md` to the command.
+
+**Remark.** The bundled Pyodide packages uses lz4 compression when producing
+`PACKAGE_NAME.data`. These instructions skip this step as it requires
+additional dependencies, which complicates the process. Further, if your
```

Comment:
The issue is that we currently don't officially support disabling the lz4 compression (which would require changing the setting in the Makefile). Disabling lz4 would need to be done on the full pyodide distribution, not just this particular package, right? Unless you are saying that this would work on a per package basis (e.g. with pyodide from the CDN)?

Reply:
The lz4 decompression happens within `PACKAGE_NAME.js`. You can mix and match uncompressed and compressed package. The goal of my changes is to use pyodide from the CDN and ship my own modules separately.